### ISSUE TITLE
OpenSSL 1.1.1 kerrcak1600-s390x.S compilation error.

### ISSUE THREAD

**Issue by likema** [openssl/openssl#7237]
When cross compiling openssl 1.1.1, it encountered:

```
crypto/sha/keccak1600-s390x.S: Assembler messages:
crypto/sha/keccak1600-s390x.S:356: Error: operand out of range (0xfffffffffffffff8 is not between 0x0000000000000000 and 0x0000000000000fff)
```

The version of `s390x-rhel4-linux-gnu-as`:

```
GNU assembler 2.15.92.0.2 20040927
Copyright 2002 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License.  This program has absolutely no warranty.
This assembler was configured for a target of `s390x-rhel4-linux-gnu'.
```

The version of `s390x-rhel4-linux-gnu-gcc`:

```
Using built-in specs.
COLLECT_GCC=s390x-rhel4-linux-gnu-gcc
COLLECT_LTO_WRAPPER=/opt/x-tools/s390x-rhel4-linux-gnu/libexec/gcc/s390x-rhel4-linux-gnu/4.8.5/lto-wrapper
Target: s390x-rhel4-linux-gnu
Configured with: ~/src/gcc/rhel4-cross/build/gcc-4.8.5/configure --target=s390x-rhel4-linux-gnu --with-sysroot=/opt/x-tools/s390x-rhel4-linux-gnu/s390x-rhel4-linux-gnu/sysroot --with-build-sysroot=~/src/gcc/rhel4-cross/dist-s390x/opt/x-tools/s390x-rhel4-linux-gnu/s390x-rhel4-linux-gnu/sysroot --with-gmp=~/src/gcc/rhel4-cross/dist-s390x/opt/x-tools/s390x-rhel4-linux-gnu --with-mpfr=~/src/gcc/rhel4-cross/dist-s390x/opt/x-tools/s390x-rhel4-linux-gnu --with-mpc=~/src/gcc/rhel4-cross/dist-s390x/opt/x-tools/s390x-rhel4-linux-gnu --with-system-zlib --enable-shared --enable-threads=posix --enable-__cxa_atexit --enable-languages=c,c++ --enable-lto --disable-multilib --disable-libstdcxx-pch --disable-nls --prefix=/opt/x-tools/s390x-rhel4-linux-gnu
Thread model: posix
gcc version 4.8.5 (GCC)
```

**Comment by levitte** [comment]
That's this line in `block_absorb`:

``` asm
	stg	%r0,-8(%r1)
```

Someone a little more s390x savvy will have to do the deeper analysis, I'm afraid.

**Comment by dot-asm** [comment]
This is fixed by #7239. Since I can reproduce it myself and confirm solution, I'm closing this.
